### ISSUE TITLE
version 8.0.0 raises ValueError when parsing tags

### ISSUE THREAD

**Issue by mazzus** [python-semantic-release/python-semantic-release#628]
## The problem

When running `semantic-release version` version 8.0.0 raises ValueError when parsing tags.
It seems that python-semantic-release assumes that all tags are in the configured format and crashes when a tag does not adhere to the format. 
In this particular instance, the issue is that the main branch has a tag with the name "tested-on-prototype".

```
semantic-release --noop version
🛡 You are running in no-operation mode, because the '--noop' flag was supplied
[13:06:26] WARNING  [semantic_release.cli.config] WARNING config.from_raw_config: the token for the remote VCS is configured as stored in the GH_TOKEN environment variable, but it is empty                  config.py:353
Traceback (most recent call last):
  File "REDACTED/bin/semantic-release", line 8, in <module>
    sys.exit(main())
  File "REDACTED/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "REDACTED/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "REDACTED/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "REDACTED/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "REDACTED/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "REDACTED/lib/python3.10/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "REDACTED/lib/python3.10/site-packages/semantic_release/cli/commands/version.py", line 274, in version
    new_version = next_version(
  File "REDACTED/lib/python3.10/site-packages/semantic_release/version/algorithm.py", line 224, in next_version
    all_git_tags_as_versions = tags_and_versions(repo.tags, translator)
  File "REDACTED/lib/python3.10/site-packages/semantic_release/version/algorithm.py", line 37, in tags_and_versions
    ts_and_vs = [(t, translator.from_tag(t.name)) for t in tags]
  File "REDACTED/lib/python3.10/site-packages/semantic_release/version/algorithm.py", line 37, in <listcomp>
    ts_and_vs = [(t, translator.from_tag(t.name)) for t in tags]
  File "REDACTED/lib/python3.10/site-packages/semantic_release/version/translator.py", line 71, in from_tag
    raise ValueError(
ValueError: Tag 'tested-on-prototype' doesn't match tag format 'v{version}'
```

## Expected behavior
The tags should be parsed and the version updated like in version 7.34.6, ignoring any tags that do not adhere to the version format.

## Environment
Python 3.10.6
pip 23.0.1
semantic-release, version 8.0.0

## Configuration

Please add your `semantic-release` configuration, and if applicable also provide
your `build-system` configuration from `pyproject.toml`.

## Logs

Logs from runnim semantic-release version:
```
semantic-release --noop -vv version 
[13:08:48] DEBUG    [semantic_release.cli.commands.main] DEBUG main.main: logging level set to: DEBUG                                                                                                           main.py:106
🛡 You are running in no-operation mode, because the '--noop' flag was supplied
           DEBUG    [semantic_release.cli.commands.main] DEBUG main.main: global cli options: GlobalCommandLineOptions(noop=True, verbosity=2, config_file='pyproject.toml', strict=False)                      main.py:124
           INFO     [semantic_release.cli.commands.main] INFO main.main: Loading TOML configuration from pyproject.toml                                                                                         main.py:128
           INFO     [semantic_release.cli.config] INFO config.select_branch_options: Using group 'main' options, as '(main|master)' matches 'main'                                                            config.py:250
           WARNING  [semantic_release.cli.config] WARNING config.from_raw_config: the token for the remote VCS is configured as stored in the GH_TOKEN environment variable, but it is empty                  config.py:353
           DEBUG    [semantic_release.changelog.template] DEBUG template.environment: {'template_dir': 'templates', 'block_start_string': '{%', 'block_end_string': '%}', 'variable_start_string': '{{',     template.py:51
                    'variable_end_string': '}}', 'comment_start_string': '{#', 'comment_end_string': '#}', 'line_statement_prefix': None, 'line_comment_prefix': None, 'trim_blocks': False,                               
                    'lstrip_blocks': False, 'newline_sequence': '\n', 'keep_trailing_newline': False, 'extensions': (), 'autoescape': True, 'autoescape_value': True}                                                      
           DEBUG    [semantic_release.version.translator] DEBUG translator._invert_tag_format_to_re: inverted tag_format 'v{version}' to 'v(?P<version>.*)'                                                translator.py:40
           DEBUG    [semantic_release.cli.commands.version] DEBUG version.is_forced_prerelease: force_prerelease = False, force_level = None, prerelease = False                                              version.py:48
Traceback (most recent call last):
  File "REDACTED/bin/semantic-release", line 8, in <module>
    sys.exit(main())
  File "REDACTED/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "REDACTED/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "REDACTED/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "REDACTED/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "REDACTED/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "REDACTED/lib/python3.10/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "REDACTED/lib/python3.10/site-packages/semantic_release/cli/commands/version.py", line 274, in version
    new_version = next_version(
  File "REDACTED/lib/python3.10/site-packages/semantic_release/version/algorithm.py", line 224, in next_version
    all_git_tags_as_versions = tags_and_versions(repo.tags, translator)
  File "REDACTED/lib/python3.10/site-packages/semantic_release/version/algorithm.py", line 37, in tags_and_versions
    ts_and_vs = [(t, translator.from_tag(t.name)) for t in tags]
  File "REDACTED/lib/python3.10/site-packages/semantic_release/version/algorithm.py", line 37, in <listcomp>
    ts_and_vs = [(t, translator.from_tag(t.name)) for t in tags]
  File "REDACTED/lib/python3.10/site-packages/semantic_release/version/translator.py", line 71, in from_tag
    raise ValueError(
ValueError: Tag 'tested-on-prototype' doesn't match tag format 'v{version}'
```

Running version 7.34.6 of python-semantic-release in the same repository:
```
semantic-release --noop -v DEBUG version 
debug: Main args: {'force_level': None, 'prerelease': False, 'prerelease_patch': True, 'post': False, 'retry': False, 'noop': False, 'define': ()}
debug: Environment: GH_TOKEN="None",GL_TOKEN="None",PYPI_PASSWORD="None",PYPI_TOKEN="None",PYPI_USERNAME="None",REPOSITORY_USERNAME="None",REPOSITORY_PASSWORD="None",
debug: Main config: {'check_build_status': False, 'commit_subject': '{version}', 'commit_message': 'Automatically generated by python-semantic-release', 'commit_parser': 'semantic_release.history.angular_parser', 'patch_without_tag': False, 'major_on_zero': True, 'upload_to_pypi': True, 'upload_to_repository': True, 'version_source': 'commit', 'no_git_tag': None}
Creating new version
Current version: 4.6.0, Current release version: 4.6.0
warning: No operation mode. Should have bumped from 4.6.0 to 4.6.1
```

**Comment by bernardcooke53** [comment]
This should be addressed in 8.0.1 :crossed_fingers:

**Comment by mazzus** [comment]
Verified that it runs correctly on the affected repo when using version 8.0.1 :+1: :clap:
